tests/Fixer/PhpUnit/PhpUnitSetUpTearDownVisibilityFixerTest.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\PhpUnit;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\PhpUnit\PhpUnitSetUpTearDownVisibilityFixer
+ */
+final class PhpUnitSetUpTearDownVisibilityFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideFixCases
+     *
+     * @param mixed      $expected
+     * @param null|mixed $input
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function provideFixCases()
+    {
+        return [
+            'setUp and tearDown are made protected if they are public' => [
+                '<?php
+class FixerTest extends \PhpUnit\FrameWork\TestCase
+{
+
+    protected function setUp() {}
+
+    protected function tearDown() {}
+}
+',
+                '<?php
+class FixerTest extends \PhpUnit\FrameWork\TestCase
+{
+
+    public function setUp() {}
+
+    public function tearDown() {}
+}
+',
+            ],
+            'setUp and tearDown stay protected if they are' => [
```

Comment:
great idea, however you already test this as part of the `expected` result in the previous test case

Reply:
True, but this test explicitly makes sure nothing changes if they are protected.
So even if it may be redundant, i personally like having that extra check. If needed i can remove it though.

Reply:
I understand, but you already have the extra test, when ever you pass;
```
        $this->doTest(
            $a,
            $b
        );
```

it will always also test;

```
        $this->doTest(
            $a,
            null
        );
```

so if you have two test with same value for `$a` than one repeats the other